### PR TITLE
Stats refactor

### PR DESCRIPTION
**Pull request by mitya52**

### CODE REVIEW

File: self_hosting_machinery/webgui/selfhost_statistics.py [hunk 1]
```diff
@@ -85,11 +77,8 @@ async def streamer():
         except Exception as e:
             raise HTTPException(status_code=500, detail=str(e))
 
-    async def _dash_prime_get(self):
-        if not self._stats_service.is_ready:
-            return self._stats_service_not_available_response
-
-        data_tables = await compose_data_frames(self._stats_service)
+    async def _dash_prime_get(self, request: Request):
+        data_tables = await self._stats_service.compose_data_frames(request)
```
Comment (valaises) [inline self_hosting_machinery/webgui/selfhost_statistics.py:81]: I dislike passing request in stats_service

File: self_hosting_machinery/webgui/selfhost_statistics.py [hunk 2]
```diff
@@ -104,11 +93,8 @@ async def _dash_prime_get(self):
             media_type='application/json'
         )
 
-    async def _dash_teams_get(self):
-        if not self._stats_service.is_ready:
-            return self._stats_service_not_available_response
-
-        data_tables = await compose_data_frames(self._stats_service)
+    async def _dash_teams_get(self, request: Request):
+        data_tables = await self._stats_service.compose_data_frames(request)
```
Comment (valaises) [inline self_hosting_machinery/webgui/selfhost_statistics.py:97]: same. dislike passing request

File: self_hosting_machinery/webgui/selfhost_statistics.py [hunk 3]
```diff
@@ -249,12 +226,16 @@ async def _telemetry_snippets(self, data: TelemetryBasicData, request: Request,
 
 
 class TabStatisticsRouter(BaseTabStatisticsRouter):
+
```
Comment (valaises) [inline self_hosting_machinery/webgui/selfhost_statistics.py:229]: extra \n. Inconsistent formatting

File: self_hosting_machinery/webgui/selfhost_statistics.py [hunk 4]
```diff
@@ -5,14 +5,14 @@
...
 from pydantic import BaseModel
 from fastapi.responses import StreamingResponse
-from fastapi import APIRouter, Request, HTTPException, Header
+from fastapi import APIRouter, Request, HTTPException
 from fastapi.responses import JSONResponse
 
 from self_hosting_machinery.dashboards.dash_prime import dashboard_prime
 from self_hosting_machinery.dashboards.dash_teams import teams_data, dashboard_teams
 from self_hosting_machinery.webgui.selfhost_database import StatisticsService, ScyllaBatchInserter
+from self_hosting_machinery.webgui.selfhost_database import StatisticsService, ScyllaBatchInserter
```
Comment (valaises) [inline self_hosting_machinery/webgui/selfhost_statistics.py:14]: the same as line 13

File: self_hosting_machinery/dashboards/utils.py [hunk 5]
```diff
@@ -1,13 +1,9 @@
...
-from typing import Dict, Any, Optional
 from dataclasses import dataclass
-
-
-from self_hosting_machinery.webgui.selfhost_database import StatisticsService
+from datetime import datetime
+from typing import Dict, Any
 
 
 @dataclass
```
Comment (valaises) [inline self_hosting_machinery/dashboards/utils.py:9]: not sure if I'd move it to selfhost_database  🤔

**Review by valaises — CHANGES_REQUESTED** [review]

File: self_hosting_machinery/webgui/selfhost_statistics.py [hunk 6]
```diff
@@ -48,30 +48,22 @@ def __init__(
...
-            status_code=500)
         self.add_api_route('/rh-stats', self._rh_stats, methods=["GET"])
         self.add_api_route("/telemetry-basic", self._telemetry_basic, methods=["POST"])
         self.add_api_route("/telemetry-snippets", self._telemetry_snippets, methods=["POST"])
         self.add_api_route('/dash-prime', self._dash_prime_get, methods=['GET'])
         self.add_api_route('/dash-teams', self._dash_teams_get, methods=['GET'])
         self.add_api_route('/dash-teams', self._dash_teams_post, methods=['POST'])
 
-    async def _account_from_bearer(self, authorization: str) -> str:
+    async def _account_from_request(self, request: Request) -> Any:
```
[the remark this answers is not in the document]
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_statistics.py]: fixed

File: self_hosting_machinery/webgui/selfhost_statistics.py (hunk 1 above)
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_statistics.py]: fixed

File: self_hosting_machinery/webgui/selfhost_statistics.py (hunk 2 above)
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_statistics.py]: fixed

File: self_hosting_machinery/webgui/selfhost_statistics.py (hunk 3 above)
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_statistics.py]: fixed

File: self_hosting_machinery/webgui/selfhost_statistics.py (hunk 4 above)
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_statistics.py]: fixed

File: self_hosting_machinery/dashboards/utils.py (hunk 5 above)
Comment (mitya52) [reply self_hosting_machinery/dashboards/utils.py:9]: ))

**Review by valaises — APPROVED** [review]

**Pull request merged**